Task 8 PR

pmvald opened the pull request:
@YusukeYoshihiro I made some comments on your task PR. Please attend to them. You can comment here any question you may have.

pmvald commented on -sass/styles/_iboxes.scss:
```diff
@@ -0,0 +1,33 @@
+    margin-top: -2rem;
+    margin-bottom: 1.5rem;
+
+    .i-some-icon{
```
apply this style to <i> element

YusukeYoshihiro replied:
Okay, I have moved it to `<i>` element.

pmvald commented on tasks/task-008/ibox.html:
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
```
remove the folder .sass-cache and add it to gitignore as well

YusukeYoshihiro replied:
I removed `.sass-cache` folder and added it to gitignore.

pmvald commented on -sass/styles/_iboxes.scss:
```diff
@@ -0,0 +1,33 @@
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin-top: 2rem;
```
never add margin to any component

YusukeYoshihiro replied:
I removed it and moved ` margin-top: 2rem;` into `<i>` element.

pmvald commented on -sass/styles/_iboxes.scss:
```diff
@@ -0,0 +1,33 @@
+  align-items: center;
+  margin-top: 2rem;
+  padding: 0 2rem 2rem;
+  border: solid 1px #555353;
```
use color variables, $grey, $light-grey, etc. see _variables in _styles folder

pmvald commented on -sass/styles/_iboxes.scss:
```diff
@@ -0,0 +1,33 @@
+.s-ibox-col{
+  display: flex;
+  flex-direction: column;
+  align-items: center;
```
Why using flex? may not be needed

YusukeYoshihiro replied:
The reason why I used `display-flex` is because I wanted to centralize the Items.
But as you suggested to me, it not necessary. So I adopt another way and apply to `<i>` element by using  `margin: auto;`

pmvald commented on -sass/styles/_iboxes.scss:
```diff
@@ -0,0 +1,33 @@
+    margin-bottom: 1.5rem;
+
+    .i-some-icon{
+      display: flex;
+      justify-content: center;
+      align-items: center;
```
No need to use flex

YusukeYoshihiro replied:
Okay got it, thank you for correcting and review my code.

pmvald reviewed (commented):
ii

YusukeYoshihiro commented on -sass/styles/_iboxes.scss:
```diff
@@ -0,0 +1,33 @@
+      font-size: 1.5rem;
+      width: 100%;
+      height: 100%;
+      border: solid 1px #555353;
```
I modified and picked up `$grey`.

pmvald commented:
@YusukeYoshihiro Good! So please push the changes so that I see. Right now there is only one commit from you.

pmvald commented:
@YusukeYoshihiro Please try again. And use this area for comments. We closed the Trello board

YusukeYoshihiro commented:
I pushed it, please confirm it.

pmvald commented:
@YusukeYoshihiro It's better now. I made some improvements as well. Please note the following:

1. I improved the ibox.html file and added more cards to that. Note that this file serves as a demo file and it is important that it has a good presentation of the component. So we have to add more content to that.
2. To add a sample text, use the well-known [Lorem Ipsum](https://www.lipsum.com/) as I did for the paragraphs.

Now your follow-up assignment is to answer the following questions (answer them here in the PR conversation:)

1. I added `height: 100` to the main class. Why? What happens if we remove it?
2. I reduced the font-size of h1 element. Why not using a smaller header like h2 or h3?
3. Why did I use primary color for the icon?

YusukeYoshihiro commented:
@pmvald Thank you for you code review.
  
My assignments' answers are below:
1.  I added height: 100 to the main class. Why? What happens if we remove it?
-> The reason is why we used `height 100` is because adjusting multiple cards' height as equally not to having different `height` depending on their volume of content. 

2. I reduced the font-size of h1 element. Why not using a smaller header like h2 or h3?
-> First of all, we need to care about `SEO` because these cards will be contained significant content `titles` or `headings` for customers(users) to boost searching our websites. So then, as the `<h1>` tag is the most important heading on the page, our CSS styling should reflect that. Furthermore, Google said that `<h1>` tags help Google to understand the structure of a page. So if we’re using `<h1>` tags as Google recommends for our page title or content heading, our `<h1>` is effectively telling Google “here’s what my page is about.” Therefore if we want to emphasize the content of the cards, we should prior to use for `<h1>` than `<h2>`, `<h3>`. However `w3school` says basically  `<h1>` should be used "only one" per page because it should represent the main heading/subject for the whole page. Also, do not skip heading levels like starting with`<h1>`, then use `<h2>`,`<h3>`. On the other hand, Google says it doesn’t matter if there are multiple `h1` tags on the website, but not to have too long sentences into `<h1>` tags. Moreover, `<h1>` tags improve user experience in the sense that they’re part of a web page’s hierarchical structure. The `<h1>` is the most prominent heading on the page, then `<h2>`, then `<h3>`, etc.
From the above, we can use `<h1>` as the titles and headings for these cards to help SEO.

3. Why did I use primary color for the icon?
-> The reason is icons should be emphasized to improve `user experience` with CSS styling by using `primary-color`.

pmvald commented:
@YusukeYoshihiro Good! For No. 1, yes, height is used to make all the cards having the same height when placed horizontally.

No 2., yes it's all about SEO and also web [semantics](https://html.spec.whatwg.org/multipage/dom.html#semantics-2). We are actually trying to decide how to use header tags. There are conflicting ideas, some say use mainly h1, h2 and some say only use h1 once and then use h3-h6 ... I'm thinking about two options:

1. We mainly use h1 for all headings and h2 for sub-heading and then not to use h3-h6
2. We use h1 only once in the page (where?) and then use h2 for all other headings (like the heading for the ibox).

I didn't have much time researching this. So please help us decide which option makes more sense by doing some research. You wanna take a look at these resources:

- [Google idea](https://www.searchenginejournal.com/heading-tags-for-seo/341817/)
- web [semantic](https://html.spec.whatwg.org/multipage/dom.html#semantics-2) Note that web semantic is about how h1,h2, etc. give meaning to your page, and it is not always inline with what Google understands from your page ...

So add your finding to the explanation you wrote for question 2, and once we made a decision I'm gonn add all these texts to the design guide so that we all follow.

For no. 3, yes, the colors and also text style (bold, italic) are used to provide different degrees of attention to the user. The important parts should stand out by making them bold/colorful/big, on the other hand, the non-important/secondary parts should be small/grey/thin font ... For these cards, we use icons to draw people attention (thus it is called ibox=icon box). So icon is the primary section that draws attention. We could have used the header instead as the main part. In this case we had to make the header colorful (or bold) and the icon grey and the component would be called hbox (=header box) instead of ibox.
